src/aspire/abinitio/commonline_c3_c4.py

```diff
@@ -633,3 +793,110 @@ def _signs_times_v(self, vijs, vec):
+
+        :param r3: A 1x3 vector of norm 1.
+        :return: A 3x3 rotation matrix whose third row is r3.
+        """
+
+        # If the third row coincides with the z-axis we return the identity matrix.
+        if norm(r3 - [0, 0, 1]) < 1e-5:
+            return np.eye(3)
+
+        # tmp is non-zero since r3 does not coincide with the z-axis.
+        tmp = np.sqrt(r3[0] ** 2 + r3[1] ** 2)
+
+        # Construct an orthogonal row vector of norm 1.
+        r1 = np.array([r3[1] / tmp, -r3[0] / tmp, 0])
+
+        # Construct r2 so that r3 = r1xr2
+        r2 = np.array([r3[0] * r3[2] / tmp, r3[1] * r3[2] / tmp, -tmp])
+
+        return np.vstack((r1, r2, r3))
+
+    @staticmethod
+    def clAngles2Ind(clAngles, n_theta):
+        thetas = np.arctan2(clAngles[:, 1], clAngles[:, 0])
+
+        # Shift from [-pi,pi] to [0,2*pi).
+        thetas = np.mod(thetas, 2 * np.pi)
+
+        # linear scale from [0,2*pi) to [0,n_theta).
+        return np.mod(np.round(thetas / (2 * np.pi) * n_theta), n_theta).astype(int)
+
+    @staticmethod
+    def g_sync(rots, order, rots_gt, dtype=np.float32):
+        """
+        Every estimated rotation might be a version of the ground truth rotation
+        rotated by g^{s_i}, where s_i = 0, 1, ..., order. This method synchronizes the
+        ground truth rotations so that only a single global rotation need be applied
+        to all estimates for error analysis.
+
+        :param rots: Estimated rotation matrices
+        :param order: The cyclic order asssociated with the symmetry of the underlying molecule.
+        :param rots_gt: Ground truth rotation matrices.
+        :param dtype: The dtype of the rotation
+
+        :return: g-synchronized ground truth rotations.
+        """
+        assert len(rots) == len(
+            rots_gt
+        ), "Number of estimates not equal to number of references."
+        n_img = len(rots)
+
+        rots_symm = cyclic_rotations(order, dtype)
+
+        A_g = np.zeros((n_img, n_img), dtype=complex)
+
+        pairs = all_pairs(n_img)
+
+        for (i, j) in pairs:
+            Ri = rots[i]
+            Rj = rots[j]
+            Rij = Ri.T @ Rj
+
+            Ri_gt = rots_gt[i]
+            Rj_gt = rots_gt[j]
+
+            diffs = np.zeros(order)
+            for s, g_s in enumerate(rots_symm):
+                Rij_gt = Ri_gt.T @ g_s @ Rj_gt
+                diffs[s] = min([norm(Rij - Rij_gt), norm(Rij - J_conjugate(Rij_gt))])
+
+            idx = np.argmin(diffs)
+
+            A_g[i, j] = np.exp(-1j * 2 * np.pi / order * idx)
+
+        # A_g(k,l) is exp(-j(-theta_k+theta_l))
+        # Diagonal elements correspond to exp(-i*0) so put 1.
+        # This is important only for verification purposes that spectrum is (K,0,0,0...,0).
+        A_g += np.conj(A_g).T + np.eye(n_img)
+
+        eig_vals, eig_vecs = eigh(A_g)
+        evect1 = eig_vecs[:, -1]
+
+        angles = np.exp(1j * 2 * np.pi / order * np.arange(order))
+        sign_g_Ri = np.zeros(n_img)
```

Comment:
Changed this to `power_g_Ri` since this represents the exponent on the rotation matrix `g` that rotates by multiples of `2 * np.pi / order`.

Also, no longer initializing this since there is no need to hold on to this value post refactor.